src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -672,71 +677,33 @@ const static auto formats = {
-    const std::string& datetime, const boost::posix_time::ptime& default_time = epoch) {
-    boost::posix_time::ptime utc_time{boost::posix_time::not_a_date_time};
-    auto millis = 0LL;
-
-    // Empty datetime is 0.
-    if (!datetime.empty()) {
-        for (const auto& format : formats) {
-            std::istringstream date_stream{datetime};
-            date_stream.imbue(format);
-            boost::local_time::local_date_time local_date{boost::local_time::not_a_date_time};
-            if (date_stream >> local_date) {
-                utc_time = local_date.utc_time();
-
-                millis = (utc_time - epoch).total_milliseconds();
-                break;
-            }
-        }
-
-        if (utc_time == boost::posix_time::not_a_date_time) {
-            try {
-                // Last gasp try to interpret unsigned long long.
-                millis = std::stoull(datetime);
-            } catch (const std::invalid_argument& _) {
-                auto msg = "^RandomDate: Invalid Dateformat '" + datetime + "'";
-                BOOST_THROW_EXCEPTION(InvalidValueGeneratorSyntax{msg});
-            }
-        }
-
-    } else {
-        millis = (default_time - epoch).total_milliseconds();
-    }
-    return std::chrono::milliseconds{millis};
-}
 
 /** `{^RandomDate: {min: "2015-01-01", max: "2015-01-01T23:59:59.999Z"}}` */
 class RandomDateGenerator : public Generator<bsoncxx::types::b_date> {
 public:
     RandomDateGenerator(const Node& node, GeneratorArgs generatorArgs)
         : _rng{generatorArgs.rng},
-          _min{parseDateAsMillis(node["min"].maybe<std::string>().value_or(""))},
-          _max{parseDateAsMillis(node["max"].maybe<std::string>().value_or(""), max_date)} {
+          _minGen{dateGenerator(node["min"], generatorArgs)},
+          _maxGen{dateGenerator(node["max"], generatorArgs, max_date)} {}
 
-        if (_max <= _min) {
-            auto max = node["max"].maybe<std::string>().value_or("");
-            auto min = node["min"].maybe<std::string>().value_or("");
-
-            std::string msg = "^RandomDate: max('" + max + " / " + std::to_string(_max.count()) +
-                "') must be greater than min('" + min + " / " + std::to_string(_min.count()) + "')";
-            BOOST_THROW_EXCEPTION(InvalidValueGeneratorSyntax{msg});
+    bsoncxx::types::b_date evaluate() override {
+        auto minValue = _minGen->evaluate();
+        auto maxValue = _maxGen->evaluate();
+        if (maxValue <= minValue) {
+            std::stringstream ex;
+            ex << "^RandomDate: max('" << maxValue << "') must be greater than min('" << minValue
+               << "')";
+            BOOST_THROW_EXCEPTION(InvalidValueGeneratorSyntax{ex.str()});
         }
-    }
 
-    bsoncxx::types::b_date evaluate() override {
-        boost::random::uniform_int_distribution<long long> dist{_min.count(), _max.count() - 1};
+        boost::random::uniform_int_distribution<long long> dist{minValue, maxValue - 1};
```

Comment:
With the extra parameterization and generators, do you want to make kinds based on all the int distribution generators?

Reply:
Yeah, keep the scope reasonable. I could imagine though having a set range (today to end of year), but not wanting that to be uniformly distributed.